Data-Structure-Linked-Lists.md

```diff
@@ -0,0 +1,403 @@
+
+```python
+
+class Node(object):
+    # Constructor
+    def __init__(self, data=None, next=None):
+        self.data = data
+        self.next = next
+
+    # Function to get data
+    def get_data(self):
+        return self.data
+
+    # Function to get next node
+    def get_next(self):
+        return self.next
+
+    # Function to set next field
+    def set_next(self, new_next):
+        self.next = new_next
+
+
+class LinkedList(object):
+    def __init__(self, head=None):
+        self.head = head
+
+    # Function to insert data
+    def insert(self, data):
+        # new_node is a object of class Node
+        new_node = Node(data)
+        new_node.set_next(self.head)
+        self.head = new_node
+        print("Node with data " + str(data) + " is created succesfully")
+
+    # Function to get size
+    def size(self):
+        current = self.head
+        count = 0
+        while current:
+            count += 1
+            current = current.get_next()
+        print("Size of link list is " + str(count))
+
+    # Function to search a data
+    def search(self, data):
+        current = self.head
+        found = False
+        while current and found is False:
+            if current.get_data() == data:
+                found = True
+            else:
+                current = current.get_next()
+        if current is None:
+            print("Node with data " + str(data) + " is not present")
+        else:
+        	print("Node with data " + str(data) + " is found")
+
+
+    # Function to delete a node with data
+    def delete(self, data):
+        current = self.head
+        previous = None
+        found = False
+        while current and found is False:
+            if current.get_data() == data:
+                found = True
+            else:
+                previous = current
+                current = current.get_next()
+        if current is None:
+            print("Node with data " + str(data) + " is not in list")
+        elif previous is None:
+            self.head = current.get_next()
+            print("Node with data " + str(data) + " is deleted successfully")
+        else:
+            previous.set_next(current.get_next())
+            print("Node with data " + str(data) + " is deleted successfully")
+
+SLL = LinkedList() # Creates an object of class LinkedList
+SLL.size() # prints 'Size of link list is 0'
+data_elements = [5, 10, 2, 6, 8, 20]
+for each in data_elements:
+	SLL.insert(each)
+SLL.size() # prints 'Size of link list is 6'
+SLL.search(4) # prints 'Node with data 4 is not present'
+SLL.search(5) # prints 'Node with data 5 is found'
+SLL.delete(4) # prints 'Node with data 4 is not in list'
+SLL.delete(5) # prints 'Node with data 5 is deleted successfully'
+```
+:rocket: [Run Code](https://repl.it/CVq3)
+
+
+**Advantages**
+
+1. Linked lists are a dynamic data structure, which can grow and shrink, allocating and deallocating memory while the program is running.
```

Comment:
Should I say automatically or destructor is called automatically due to which memory is collected in python and java...?